oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Checkpoints.java

```diff
@@ -118,7 +118,7 @@ public void release(String checkpoint) {
      */
     @SuppressWarnings("unchecked")
     @Nullable
-    public Revision getOldestRevisionToKeep() {
+    public Revision getOldestRevisionToKeep(boolean performCleanup) {
```

Comment:
Would suggest to add a javadoc description of the new `@param`

Reply:
IMO, a better name would be `isReadOnlyRepo`

Reply:
I'm adding the `@param`. 

But I don't think we should change the name to `isReadOnlyRepo` because then it may be misleading. We could know in this method if the repository is read-only simply calling:
`nodeStore.isReadOnlyMode();`

In our use-case we want to avoid the checkpoint cleanup when the repository is read-only, but I don't think we should do that in general because we would be changing the logic of other places where this method is already used.